geo/src/algorithm/offset/cross_product.rs

```diff
@@ -0,0 +1,85 @@
+use crate::CoordFloat;

```

Comment:
I'm a bit confused about this. Would the [dot product sign](https://docs.rs/geo/latest/geo/algorithm/kernels/trait.Kernel.html#method.dot_product_sign), i.e. the [Orientation](https://docs.rs/geo/latest/geo/algorithm/kernels/enum.Orientation.html) suffice? It looks like you're just checking whether the result is `-1`, `1`, or `0`…

Reply:
Apologies, looks like my inept git-fu has disconnected all your comments from the code 😕 I think maybe re-basing the branch on my fork was the wrong thing to do.

`cross_product` simplifies `line_segment_intersection_with_parameters()`: While it is used for a collinear check at the start of that function (which could be replaced with `Kernel::dot_product_sign` or `Kernel::orient2d`) it is used twice more further along to calculate the parameters `t_ab` and `t_cd` so I do not believe it can be substituted for an orientation check.

```rust
let t_ab = cross_product_2d(ac, cd) / ab_cross_cd;
let t_cd = -cross_product_2d(ab, ac) / ab_cross_cd;
```

`line_segment_intersection_with_parameters()` is currently only used via a wrapper function `line_segment_intersection_with_relationships()`; in that function it checks each `t_ab` and `t_cd` to find which of three ranges they lie in:

- less than zero
- between zero and one
- greater than one

It's possible there is some tricky way to express this check in terms of `orient2d()`, I'll have a think about it.

It does feel weird to have `cross_product()` out in it's own module. In the past I have worked with libraries that have a `Vector2` type where this kind of function would be implemented. It is simple enough to be inline perhaps, but I wrote up all the tests and documentation because in the past I have made many mistakes with putting negatives in the wrong place and if you make this kind of mistake an even number of times you might not realize since it cancels out 😋